src/main/kotlin/com/hypto/iam/server/service/UsersService.kt

```diff
@@ -219,8 +220,12 @@ class UsersServiceImpl : KoinComponent, UsersService {
     }
 
     override suspend fun authenticate(username: String, password: String): User {
-        val userRecord = userRepo.findByAliasUsername(username)
-            ?: throw AuthenticationException("Invalid username and password combination")
+        val userRecord = if (EMAIL_REGEX.toRegex().matches(username)) {
+            userRepo.findByEmail(username)
```

Comment:
Can preferred username ever match with email regex ? If so, this condition can break.

Reply:
preferred_username accepts only Alphanumeric and Underscore. So, email can't be set as username. This way need not even query the preferred_username column.